File: labs/M01-llm-mental-model/solution/model_comparison.js
/**
 * M01 Lab - Step 3: Model Comparison — SOLUTION
 * ================================================
 * Complete working implementation.
 */

import Anthropic from "@anthropic-ai/sdk";
import "dotenv/config";

const client = new Anthropic();

/**
 * Call a specific Claude model and return the response text and elapsed time.
 * @param {string} modelName - The model identifier
 * @param {string} prompt - The prompt to send
 * @returns {Promise<[string, number]>} [response_text, elapsed_seconds]
 */
async function callModel(modelName, prompt) {
  const start = performance.now();
  const response = await client.messages.create({
    model: modelName,
    max_tokens: 1024,
    messages: [{ role: "user", content: prompt }],
  });
  const elapsed = (performance.now() - start) / 1000;
  return [response.content[0].text, elapsed];
}

async function main() {
  const prompt = "Explain what a UCC filing is in 2-3 sentences.";
  const models = [
    "claude-haiku-4-5-20251001",
    "claude-sonnet-4-6",
  ];

  console.log("--- Model Comparison ---\n");
  for (const model of models) {
    console.log(`Model: ${model}`);
    try {
      const [text, elapsed] = await callModel(model, prompt);
      console.log(`Time: ${elapsed.toFixed(2)}s`);
      console.log(`Response:\n  ${text}\n`);
    } catch (e) {
      console.log(`  [ERROR] ${e.message}\n`);
    }
  }
}

main().catch(console.error);

File: labs/M01-llm-mental-model/starter/model_comparison.py
"""
M01 Lab - Step 3: Model Comparison
====================================
Compare Claude Haiku and Sonnet on the same prompt.
"""
import time
import anthropic
from dotenv import load_dotenv

load_dotenv()
client = anthropic.Anthropic()


def call_model(model_name: str, prompt: str) -> tuple[str, float]:
    """Call a specific Claude model and return (response_text, elapsed_seconds)."""
    # TODO:
    # 1. Record start time with time.time()
    # 2. Call client.messages.create() with:
    #    - model=model_name
    #    - max_tokens=1024
    #    - messages: a single user message with the prompt
    # 3. Record end time
    # 4. Return (response_text, elapsed_time)
    pass


if __name__ == "__main__":
    prompt = "Explain what a UCC filing is in 2-3 sentences."
    models = [
        "claude-haiku-4-5-20251001",
        "claude-sonnet-4-6",
    ]

    print("--- Model Comparison ---\n")
    for model in models:
        print(f"Model: {model}")
        try:
            text, elapsed = call_model(model, prompt)
            print(f"Time: {elapsed:.2f}s")
            print(f"Response:\n  {text}\n")
        except Exception as e:
            print(f"  [ERROR] {e}\n")
